src/main/resources/org/biouno/unochoice/DynamicReferenceParameter/index.jelly

```diff
@@ -67,11 +67,11 @@
     UnoChoice.renderDynamicRenderParameter('#${paramName}', '${h.escape(it.getName())}', '${h.escape(paramName)}', referencedParameters, dynamicReferenceParameter);
 
     // update spinner id
-    let rootElmt = document.querySelector('#${paramName}');
+    var rootElmt = document.querySelector('#${paramName}');
     if (rootElmt) {
-      let divElmt = rootElmt.querySelector('div');
+      var divElmt = rootElmt.querySelector('div');
       if (divElmt) {
-        let spinnerId = divElmt.id.split('_').pop();
+        var spinnerId = divElmt.id.split('_').pop();
```

Comment:
Hi @c3p0-maif ! :wave: 

I was following the interaction in the JIRA issue, thanks for the quick fix.

I am curious why we had to revert to `var` here. My understanding is that `var` is bad practice as it pollutes other contexts, while with `const` and `let` you are supposed to properly isolate the context of each variable.

e.g.

```js
function test() {
  if (1 > 0) {
    let a = 1
  }
  console.log(a) 
}
```

prints `undefined`.

```js
function test() {
  if (1 > 0) {
    var a = 1
  }
  console.log(a) 
}
```

prints `1`.

I believe what's considered best practice is to use `let` in this case, if you really want `a` being used outside the `if` statement.

```js
function test() {
  let a
  if (1 > 0) {
    a = 1
  }
  console.log(a) 
}
```

I haven't checked the code yet, but maybe you could confirm if we are doing the same here? If so it'd be good if we could fix that too.

Thanks again! (I'm on holidays, so slower than normal to reply, sorry).

Reply:
Hi @kinow , no need to thank for my reactivity, my letmotiv : _you broke it, you fix it :)_  

I am not a javascript expert but what I see in the console log is that I cannot do twice `let a` but I can do twice `var a`.  
![image](https://github.com/jenkinsci/active-choices-plugin/assets/69896961/d3265550-06a6-441b-a1e9-107ee7727124)

I think it's why some lines upper we have an other var defined.
``` javascript
var referencedParameters = Array();
```

As I understood, this jelly code snippet is merged with other jelly snippets and built in order to have a html component.
So if 2 dynamic reference parameter are created, it "import" the jelly 2 times and so in the rendered html page we have 2 times the `let a` which throw this exception.

Reply:
>Hi @kinow , no need to thank for my reactivity, my letmotiv : you broke it, you fix it :)

:+1: :bow: 

>I am not a javascript expert but what I see in the console log is that I cannot do twice let a but I can do twice var a.
>...
> So if 2 dynamic reference parameter are created, it "import" the jelly 2 times and so in the rendered html page we have 2 times the let a which throw this exception.

Ah, interesting! I think you nailed it. I will have time to review and merge a new release between tonight and tomorrow (more likely tomorrow).

Thank you!
Bruno

Reply:
Your analysis is spot-on! Sorry for pushing for the `let`, I will keep that in mind too.